add actions to state-badge element and state-label-badge (default Lovelace badge type)

iantrich opened the pull request:
Docs: https://github.com/home-assistant/home-assistant.io/pull/10915

iantrich committed:
add actions to state-badge element

iantrich commented:
will wait for https://github.com/home-assistant/home-assistant.io/pull/10834 to be merged before adding docs

bramkragten commented on src/components/entity/ha-state-label-badge.ts:
```diff
@@ -35,6 +35,8 @@ export class HaStateLabelBadge extends LitElement {
 
   @property() public image?: string;
 
+  @property() public handleClick?: boolean;
```
```
  @property({type : Boolean}) public dontHandleClick;
```

bramkragten commented on src/components/entity/ha-state-label-badge.ts:
```diff
@@ -92,12 +94,15 @@ export class HaStateLabelBadge extends LitElement {
-      }
-    });
+
+    if (this.handleClick !== false) {
```
```
    if (this.dontHandleClick) {
        return;
    }
```

bramkragten commented on src/panels/lovelace/elements/hui-state-badge-element.ts:
```diff
@@ -61,9 +64,28 @@ export class HuiStateBadgeElement extends LitElement
+        .longPress=${longPress({
+          hasDoubleClick: hasDoubleClick(this._config!.double_tap_action),
+        })}
+        .handleClick=${false}
```
```suggestion
        dontHandleClick
```

bramkragten commented:
How many places use `HaStateLabelBadge`? Rather have 1 logic instead of 2, either have them all handle the clicks themselves or have `HaStateLabelBadge` deal with it.

iantrich commented:
> How many places use `HaStateLabelBadge`? Rather have 1 logic instead of 2, either have them all handle the clicks themselves or have `HaStateLabelBadge` deal with it.

Agreed

iantrich committed:
address comments

state-label-badge no longer handles clicks
added actions to hui-state-label-badge
moved ha-badges-card to Lit

iantrich commented:
Went with a different approach and as suggested have the parent handling click events. I realized 3/4 the way through moving ha-badges-card to Lit that it was part of States, :man_shrugging: 

In doing this I also decided to add actions to badges :tada:
